packages/core/src/lib/connection_manager.ts

```diff
@@ -164,21 +217,50 @@ export class ConnectionManager {
     );
   }
 
+  private async attemptDial(peerId: PeerId): Promise<void> {
+    if (this.currentActiveDialCount >= this.options.maxParallelDials) {
+      this.pendingPeerDialQueue.push(peerId);
+      return;
+    }
+
+    if (!(await this.shouldDialPeer(peerId))) return;
+
+    this.dialPeer(peerId).catch((err) =>
```

Comment:
This method does not throw so better way would be to just `await`/`try` and remove `catch` where the method is used.